R/field-sheet.R

```diff
@@ -182,19 +181,25 @@ get_field_sheet_event_plan <- function(con, sample_event_id_arg) {
 #' 2022_sample_plan <- add_sample_plan(con, sample_plan_2022_final, verbose = TRUE)
 #'
 #' # create workbook with field sheets for all sample event IDs in the 2022 sample plan
-#' create_multiple_field_sheets(added_sample_plan = 2022_sample_plan, "data-raw/2022_field_sheets.xlsx")
+#' create_multiple_field_sheets(con, added_sample_plan = 2022_sample_plan, "data-raw/2022_field_sheets.xlsx")
 #' @export
 #' @family field sheet helpers
 #' @md
-create_multiple_field_sheets <- function(added_sample_plan, field_sheet_filepath) {
+create_multiple_field_sheets <- function(con, season, field_sheet_filepath) {
   # create workbook to append each sampling event tab
   wb <- openxlsx::createWorkbook()
 
+  # get season sample_events
+  sample_event_ids <- dplyr::tbl(con, "sample_event") |>
+    dplyr::filter(lubridate::year(first_sample_date) == season) |>
+    dplyr::collect() |>
+    dplyr::pull(id)
+
   # loop through unique sample event IDs (input to get_field_sheet_event_plan) to append
   # workbooks
-  unique_sample_ids <- unique(added_sample_plan$sample_ids_created$sample_event_id)
+  unique_sample_event_ids <- unique(sample_event_ids)
 
-  for(i in unique_sample_ids){
+  for(i in unique_sample_event_ids){
```

Comment:
can you convert this to purr::walk, but add a progress bar, since it takes a while it would be nice to give user feedback that something is working....I think this works, but double check.

```r
purrr::walk(unique_sample_event_ids, function(i) {

plan <- get_field_sheet_event_plan(con, sample_event_id = i)
    # append a field sheet to the workbook for that sample event
    # create field sheet for sampling crews to use.
    # Takes in get_field_sheet_sample_plan$field_sheet_sample_plan
    # leaves columns "Date", "Time", "FL(mm)", "Field run ID", "Fin clip (Y/N)", and "Comments"
    # blank intentionally so that they can be filled out in the field
    wb <- create_field_sheet(wb = wb,
                             field_sheet_sample_plan = plan$field_sheet_sample_plan,
                             sample_event_number = plan$sample_event_number,
                             first_sample_date = plan$first_sample_date,
                             sample_location = plan$location_name,
                             sample_location_code = plan$location_code)
},
  .progress = list(
    type = "iterator",
    clear = TRUE,
    name = "inserting threshold result into database"
  ))
```

Reply:
done!